a couple of IPv6 packetdrill core fixes

dcaratti opened the pull request:
@mattbe, please review but wait for merging - i need to ensure that existing tests are not broken by this patch

dcaratti committed:
parser.y: don't assume IPv4 for ADD_ADDR where address is UNDEFINED

Link: https://github.com/multipath-tcp/mptcp_net-next/issues/143
Signed-off-by: Davide Caratti <dcaratti@redhat.com>

matttbe commented on gtests/net/packetdrill/parser.y:
```diff
@@ -2231,26 +2231,35 @@ tcp_option
+						TCPOLEN_ADD_ADDR_V4_PORT_HMAC : TCPOLEN_ADD_ADDR_V6_PORT_HMAC);
 			$$->data.add_addr.ipv4_w_port_hmac.port = htons($4);
-                        $$->data.add_addr.ipv4_w_port_hmac.hmac = htobe64($5.hash);
+			$$->data.add_addr.ipv4_w_port_hmac.hmac = htobe64($5.hash);
```
I guess here for the two last lines, you need to check if you have v4 or v6, no?

dcaratti replied:
> I guess here for the two last lines, you need to check if you have v4 or v6, no?

right, I'll fix this in the respin

matttbe reviewed (changes requested):
thx for looking at that :-)

matttbe commented:
@dcaratti is it linked to https://github.com/multipath-tcp/mptcp_net-next/issues/143 ?
If yes, may you add a line about that in the commit line + copied it in the first message of the PR if this is going to close the issue please? :)

dcaratti commented:
> @dcaratti is it linked to [multipath-tcp/mptcp_net-next#143](https://github.com/multipath-tcp/mptcp_net-next/issues/143) ?
> If yes, may you add a line about that in the commit line + copied it in the first message of the PR if this is going to close the issue please? :)

sure, thanks for spotting this: I will add a 'Link:' line in the commit message. Not a 'Closes:', unfortunately, because at least for servers we still need to discriminate IPv4 from IPv6 when parsing outbound `ADD_ADDR + HMAC` (the problem is similar to the one recently [discussed](https://lore.kernel.org/mptcp/aeb4b7f3044e2f869ae026f950eb70c4398dad63.1616620119.git.dcaratti@redhat.com/) upstream, and it might need a similar solution (if it's feasible)

dcaratti committed:
fix parser of IPv6 address in ADD_ADDR

the argument of ipv6_parse() should be the content of STRING, not
INET6_ADDR: fix 'add_addr' to allow specifying IPv6 addresses in
ADD_ADDR packets.

Signed-off-by: Davide Caratti <dcaratti@redhat.com>

matttbe commented on gtests/net/packetdrill/parser.y:
```diff
@@ -1875,7 +1875,7 @@ add_addr_ip
 	struct endpoint endpoint;
 	memset(&endpoint, 0, sizeof(endpoint));
-	struct ip_address ip_formatted = ipv6_parse($5);
+	struct ip_address ip_formatted = ipv6_parse($7);
```
may you add a small explanation in the commit message if you don't mind?

It looks good to me when looking at what is done in IPv4. I guess this was not validated before.

dcaratti replied:
> may you add a small explanation in the commit message if you don't mind?
> 
> It looks good to me when looking at what is done in IPv4. I guess this was not validated before.

:-) ok, I assumed the patch context was self-explainatory:
```
| ADDR '[' WORD '=' INET6_ADDR '(' STRING ')' ']' {
   $1   $2  $3   $4    $5       $6   $7
```
Sure, I'll add a line. I think it's just a typo, and probably it never worked since the beginning

matttbe replied:
thx!

> I think it's just a typo, and probably it never worked since the beginning

That's what I understood when reading the patch but it was "strange". Strange the code was there but never validated (but maybe not that strange here :) )

matttbe commented:
(thx for the update! :) )

matttbe reviewed (approved):
thx for the update!

Please tell me when I can apply your patch `mptcp: drop all sub-options except ADD_ADDR when the echo bit is set` in the kernel repo ;-)
